portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +36,59 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Generates a randome quote from the ArrayList Quotes and converts the quote to a JSON object.*/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jason</h1>");
+    Query query = new Query("Questions").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Questions> questions = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String question = (String) entity.getProperty("question");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Questions userQuestion = new Questions(name, question, timestamp);
+      questions.add(userQuestion);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(questions));
+  }
+
+  /** Posts the users questions on a message board. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String name = request.getParameter("author");
+    String question = request.getParameter("description");
+    long timestamp = System.currentTimeMillis();
+
+    Entity questionEntity = new Entity("Questions");
+    questionEntity.setProperty("name", name);
+    questionEntity.setProperty("question", question);
+    questionEntity.setProperty("timestamp", timestamp);
```

Comment:
nit: can move this into a helper function like, buildQuestionEntity(request)

Reply:
Create a new helper function?